core/src/consensus/tendermint/worker.rs

```diff
@@ -667,15 +668,17 @@ impl Worker {
                 self.request_messages_to_all(vote_step, &BitSet::all_set() - &self.votes_received);
                 if !self.already_generated_message() {
                     let block_hash = match &self.last_two_thirds_majority {
-                        TwoThirdsMajority::Empty => None,
-                        TwoThirdsMajority::Unlock(_) => None,
-                        TwoThirdsMajority::Lock(locked_view, block_hash) => {
-                            if locked_view == &self.view {
-                                Some(*block_hash)
-                            } else {
-                                None
-                            }
+                        TwoThirdsMajority::Lock(locked_view, block_hash)
```

Comment:
@majecty Would you check it?
I think tendermint should pre-commit the locked block regardless of the block generation time.

Reply:
I think it's OK.
1. The locked proposal will receive pre-commits in next round. No problem for the liveness.
2. Other nodes don't know whether a node received enough pre-votes in the prevote timeout or not. It is not the violation of the rule in the other node's view.

Reply:
On second thought, do not sending pre-vote message will be simpler.

Reply:
@sgkim126 Is it OK to send nil message in the prevote step rather than the precommit step?

Reply:
I agree with @majecty.
Prevent sending prevote would be simpler.
Please change the commit and issue.

Reply:
I changed the design following the above discussion and changed also the pr message, commit message and issue message.